GLTFSDK/Source/MeshPrimitiveUtils.cpp

```diff
@@ -721,12 +552,7 @@ std::vector<float> MeshPrimitiveUtils::GetPositions(const Document& doc, const G
         throw GLTFException("Invalid type for positions accessor " + positionsAccessor.id);
     }
 
-    if (positionsAccessor.componentType != COMPONENT_FLOAT)
-    {
-        throw GLTFException("Invalid component type for positions accessor " + positionsAccessor.id);
-    }
-
-    return reader.ReadBinaryData<float>(doc, positionsAccessor);
+    return reader.ReadFloatData(doc, positionsAccessor);
 }
 
 std::vector<float> MeshPrimitiveUtils::GetPositions(const Document& doc, const GLTFResourceReader& reader, const MeshPrimitive& meshPrimitive)
```

Comment:
I don't like that the functions all now succeed quietly when reading component types not supported by the core spec. I would prefer the new behaviour to be opt-in. Either by passing a bool (isUnsupportedComponentTypeAllowed?) or enum value optional function parameter.

Also, should COMPONENT_UNSIGNED_INT data be allowed? If not then it would be better for this function to throw an exception with a very specific message rather then get the more generic exception generated in ReadFloatData.

Reply:
COMPONENT_UNSIGNED_INT shouldn't be allowed. I want to discuss the error handling more in general before adding a more precise error message for this.